src/gsy_e/models/myco_matcher/myco_internal_matcher.py

```diff
@@ -32,14 +33,19 @@ class MycoInternalMatcher(MycoMatcherInterface):
     def __init__(self):
         super().__init__()
         self.match_algorithm = None
+        self.match_algorithm_day_ahead = None
```

Comment:
I think this might complicate the interface and make it less generic. 
AFAIU, the purpose of the creator of this interface is to always have the matching algorithm inside `self.match_algorithm`, regardless of the type of market we're using. 
The choice happens in `self.get_matching_algorithm()`, so my assumption is that that's where we could set the `PayAsClearMatchingAlgorithm` in case of day-ahead markets.

Doing so would also make `_get_matches_recommendations_day_ahead` unnecessary.

Reply:
You are right, this step is not needed, as well as `_get_matches_recommendations_day_ahead`. I think I wanted to be future prove in case we want to add logic in case we do not want to always use PayAsClear for the DayAhead market. For not it is not needed though and I will do as you propose.

Reply:
https://github.com/gridsingularity/gsy-e/pull/1449/commits/4609637ad73e2915585b015e2b26f9403b8961cf